documentation/managing_modules.md

```diff
@@ -0,0 +1,183 @@
+  need to pin a module to a specific version, avoid editing the `modules` list
+  directly. Instead, use the `bolt module add` command.
+- **Puppetfile:** The Puppet file is now a lock file. Bolt generates a
+  Puppetfile each time you modify your modules with a Bolt command. Do not edit
+  the Puppetfile directly. Instead, use Bolt commands to manage your modules,
+  and rely on Bolt to manage the Puppetfile. You can compare a Puppetfile to a
+  `Gemfile.lock` file in Ruby, or a `yarn.lock` file in Yarn.
+- **Module path:** The new module path is `['modules']` and Bolt always appends
+  `.modules` to the module path.
+
+## How it works
+
+When you install a Forge module, Bolt 
+- installs the latest version of the module along with any dependencies required
+  by the module.
+- adds entries for the module to the `modules` key in your `bolt-project.yaml`,
+  and installs the module and its dependencies into a `.modules` directory in
+  your project. 
+- generates a Puppetfile, which lists all of a project's required modules and
+  their versions.
+
+For example, if you create a project named `example_project` and install the
+`apache` module, the `modules` key in your `bolt-project.yaml` file looks
+something like this:
+
+```yaml
+name: example_project
+modules:
+- name: puppetlabs-apache
+```
+
+The project's Puppetfile lists the `apache` module and all of its
+dependencies:
+
+```puppet
+# This Puppetfile is managed by Bolt. Do not edit.
+mod "puppetlabs-apache", "5.5.0"
+mod "puppetlabs-stdlib", "6.4.0"
+mod "puppetlabs-concat", "6.2.0"
+mod "puppetlabs-translate", "2.2.0"
+```
+
+And the `.modules` directory contains all of the installed modules:
+
+```shell
+$ ls modules
+apache    concat    stdlib    translate
+```
+
+## Opting in
+
+Use `bolt project migrate` to migrate a Bolt project to the new workflows. 
+
+Bolt reads your Puppetfile and prompts you to choose which modules are direct
+dependencies of your project. Only select a module as a direct dependency if
+your project uses content from that module.
+
+After you've selected your project's direct dependencies, Bolt adds those
+modules to your `bolt-project.yaml` file and pins them to their current
+versions, installs those modules and their dependencies into a `.modules`
+directory, and generates a Puppetfile. 
+
+If you don't want your modules pinned to a specific version, edit out the
+`version_requirement` fields for each module in your `bolt-project.yaml` file
+and run `bolt module install`.
+
+> **NOTE:** Migrating a project makes irreversible changes to the project's
+  configuration and inventory files. Before you migrate, make sure the project
+  has a backup or uses a version control system.
+
+## Install modules in a Bolt project
+
+Once you've opted in to the experimental workflows, you can use the `bolt module
+add` and `bolt module install` commands to manage your Forge modules. Git
+modules are not currently supported.
```

Comment:
```suggestion
add` and `bolt module install` commands to manage your Forge modules.

> **Note:** Bolt only supports managing modules from the public Puppet
> Forge. Git modules and private Forge modules are not supported.
```

This is a fairly important point that might trip up more advanced users. Having it in a blockquote should hopefully make it a bit more noticeable.